File: src/main/java/com/fireblazerrr/survivorbot/discord/ReadyListener.java
package com.fireblazerrr.survivorbot.discord;

import com.fireblazerrr.survivorbot.SurvivorBot;
import net.dv8tion.jda.core.events.ReadyEvent;
import net.dv8tion.jda.core.hooks.ListenerAdapter;

public class ReadyListener extends ListenerAdapter {

    public ReadyListener() {
    }

    @Override
    public void onReady(ReadyEvent event) {
        SurvivorBot.info("*** Discord bot armed ***");
        event.getJDA().getGuilds().forEach(guild -> {
            SurvivorBot.info("===== " + guild.getName() + " [" + guild.getId() + "] =====");
            SurvivorBot.info("///// Text Channels /////");
            guild.getTextChannels().forEach(textChannel -> SurvivorBot.info("      " + textChannel.getName() + " [" + textChannel.getId() + "]"));
            SurvivorBot.info("///// Voice Channels /////");
            guild.getVoiceChannels().forEach(voiceChannel -> SurvivorBot.info("      " + voiceChannel.getName() + " [" + voiceChannel.getId() + "]"));
            SurvivorBot.info("///// Roles /////");
            guild.getRoles().forEach(role -> SurvivorBot.info("      " + role.getName() + " [" + role.getId() + "]"));
            SurvivorBot.info("///// Users /////");
            guild.getMembers().forEach(member -> SurvivorBot.info("      " + member.getEffectiveName() + " [" + member.getUser().getId() + "]"));
            SurvivorBot.info("==========================================================================================");
        });
    }
}

File: src/main/kotlin/com/fireblazerrr/survivorbot/discord/ReadyListener.kt
package com.fireblazerrr.survivorbot.discord

import net.dv8tion.jda.core.entities.*
import net.dv8tion.jda.core.events.ReadyEvent
import net.dv8tion.jda.core.hooks.ListenerAdapter
import java.util.logging.Logger

class ReadyListener : ListenerAdapter(){

    val logger: Logger = Logger.getLogger("Minecraft")

    override fun onReady(event: ReadyEvent) {
        logger.info("*** Discord Bot Ready ***")
        event.jda.guilds.forEach { guild: Guild ->
            logger.info("===== ${guild.name} [${guild.id}] =====")
            logger.info("///// Text Channels /////")
            guild.textChannels.forEach { textChannel: TextChannel -> logger.info("     ${textChannel.name} [${textChannel.id}]") }
            logger.info("///// Voice Channels /////")
            guild.voiceChannels.forEach { voiceChannel: VoiceChannel -> logger.info("     ${voiceChannel.name} [${voiceChannel.id}]") }
            logger.info("///// Roles /////")
            guild.roles.forEach { role: Role -> logger.info("     ${role.name} [${role.id}]")}
            logger.info("///// Users /////")
            guild.members.forEach { member: Member -> logger.info("     ${member.nickname}(${member.effectiveName} [${member.user.id}])") }
            logger.info("================================================================================")
        }
    }
}